vendor/github.com/coreos/ign-converter/translate/v33tov32/v33tov32.go

```diff
@@ -47,7 +48,13 @@ func translateRaid(old old_types.Raid) (ret types.Raid) {
 func translateLuks(old old_types.Luks) (ret types.Luks) {
 	tr := translate.NewTranslator()
 	tr.AddCustomTranslator(translateClevis)
-	tr.Translate(old.Clevis, &ret.Clevis)
+
+	// this goes from "not pointer" to "pointer" so we need to populate it
+	// if old.Cievis isn't empty and we have something to translate in
+	if !reflect.DeepEqual(old.Clevis, old_types.Clevis{}) {
+		ret.Clevis = &types.Clevis{}
+		tr.Translate(&old.Clevis, ret.Clevis)
+	}
 	tr.Translate(&old.Device, &ret.Device)
 	tr.Translate(&old.KeyFile, &ret.KeyFile)
```

Comment:
@bgilbert @prestist  I hit kind of a snag here with how Clevis gets down-translated from 3.3 to 3.2, seeing that some of the 3.2 fields are pointers but those same fields in 3.3 are actual structs. (i was initially getting "Translate needs to be called on pointers" panics but when I adjusted it to account for just that, I was getting a panic anyway because the "target" pointer in the 3.2 structure was nil by default) 

I tried to work around it like this in...a way that seemed reasonable after poking around the rest of the translator for precedent, but on the whole I'm still probably ignorant of the rest of your code base and general practices/philosophy. 

Just looking for some guidance -- I can PR something like this to the converter if a fix along these lines is acceptable, or you can tell me how I'm using it wrong, etc :smile:

Reply:
I am taking a quick look now :)

Reply:
@jkyros could we get a pr upstream with this so you can just revendor?

Reply:
Yep, thanks! Really appreciate you taking a look, I've opened a PR: https://github.com/coreos/ign-converter/pull/47

Reply:
The converter doesn't really have practices or philosophy; it's pretty hacky.  :slightly_smiling_face:

Reply:
https://github.com/coreos/ign-converter/pull/47 merged!

Reply:
Thank you both for your time on this, I very much appreciate it! :smile: